dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/DependencyGraphBuilder.java

```diff
@@ -282,9 +273,22 @@ private static void levelOrder(
             dependencyNode =
                 resolveCompileTimeDependencies(dependencyNode.getArtifact(), includeProvidedScope);
           } catch (DependencyResolutionException ex) {
-            logger.severe("Error resolving " + dependencyNode + " under " + parentNodes
-                + " Exception: " + ex.getMessage());
-            throw ex;
+            // A dependency may be unavailable any more. For example:
+            // com.google.guava:guava-gwt:jar:20.0 (compile) has transitive dependency to
+            // org.eclipse.jdt.core.compiler:ecj:jar:4.4RC4 (not found in Maven central)
```

Comment:
Detailed transitive dependency from guava to ecj:

```
            //   com.google.guava:guava-gwt:jar:20.0 (compile)
            //     com.google.gwt:gwt-dev:jar:2.8.0 (provided)
            //       org.eclipse.jetty:apache-jsp:jar:9.2.14.v20151106 (compile)
            //         org.mortbay.jasper:apache-jsp:jar:8.0.9.M3 (compile)
            //           org.apache.tomcat:tomcat-jasper:jar:8.0.9 (compile, optional:true)
            //             org.eclipse.jdt.core.compiler:ecj:jar:4.4RC4 (not found in Maven central)
```

Reply:
There's both optional and provided in this path. That's likely significant, and we probably need to account for that somehow.

Reply:
I added a logic to skip the problematic dependency if there're parents of optional:true and scope:provided.
Are you ok with the logic?